src/wallet/walletdb.cpp

```diff
@@ -540,30 +540,45 @@ static LoadResult LoadRecords(CWallet* pwallet, DatabaseBatch& batch, const std:
     return LoadRecords(pwallet, batch, key, prefix, load_func);
 }
 
+bool HasLegacyRecords(CWallet* pwallet)
+{
+    const auto& batch = pwallet->GetDatabase().MakeBatch();
+    return HasLegacyRecords(pwallet, *batch);
+}
+
+bool HasLegacyRecords(CWallet* pwallet, DatabaseBatch& batch)
+{
+    for (const auto& type : DBKeys::LEGACY_TYPES) {
+        DataStream key;
+        DataStream value{};
+        DataStream prefix;
+
+        prefix << type;
+        std::unique_ptr<DatabaseCursor> cursor = batch.GetNewPrefixCursor(prefix);
+        if (!cursor) {
+            // Could only happen on a closed db, which means there is an error in the code flow.
+            pwallet->WalletLogPrintf("Error getting database cursor for '%s' records", type);
+            throw std::runtime_error(strprintf("Error getting database cursor for '%s' records", type));
```

Comment:
Throwing a runtime error here now instead of returning `DBErrors::CORRUPT` that the older code was doing makes sense to me. In case of loading wallet, this will be caught later, ultimately evaluating to the same `DBError`:
https://github.com/bitcoin/bitcoin/blob/cfa7f70f6c99680851ab0c0df27baf9f7508747b/src/wallet/walletdb.cpp#L1219-L1223

And in case of migrating wallet flow, this will not be caught but bubble up all the way to the RPC response, which seems ok to me.